src/Microsoft.VisualStudio.Threading/AsyncReaderWriterLock.cs

```diff
@@ -639,6 +639,16 @@ protected virtual Task OnBeforeExclusiveLockReleasedAsync()
             }
         }
 
+        /// <summary>
+        /// Get the task scheduler to execute the continuation when the lock is aquired.
+        ///  AsyncReaderWriterLock uses a special <see cref="SynchronizationContext"/> to handle execusive locks, and will ignore task scheduler provided, so this is only used in a Reader lock scenario.
```

Comment:
spelling: execusive -> exclusive; aquired -> acquired
Also, the restriction of just read locks I think should be more explicit. The first line of this summary might say:

> when the **read** lock is acquired.

Also I think it would probably help drive the point home if we gave the method a matching name: `GetTaskSchedulerForReadLockRequest`.

We should also document that (I assume) we don't use this scheduler if the caller is already on what we consider to be an acceptable thread and the read lock is available immediately. Is that even true? I am not sure.

Reply:
I understand.  The question is whether there is a way to decouple it (thread management vs concurrent control)...

Reply:
Those two concerns seem highly related. Concurrency is very much a thread concern. I don't know how we can give up scheduling without giving up concurrency control.